Source/main.cpp

```diff
@@ -434,66 +461,98 @@ static void MX_GPIO_Init(void)
 
 
 /* USER CODE BEGIN 4 */
-static void read_ADC(
-            float32_t* data_hyd_port, 
-            float32_t* data_hyd_starboard,
-            float32_t* data_hyd_stern){
-  for(int i = 0; i < DSP_CONSTANTS::WORKING_BUFFER_LENGTH; i++){
-    // This is inefficient. Might be better to use the DMA to 
-    // read continously... 
-
-    // HAL_ADC_Start(&hadc);
-    // HAL_ADC_PollForConversion(&hadc, 5);
-    // adc_6 = HAL_ADC_GetValue(&hadc);
-    // HAL_ADC_Start(&hadc)
-    // HAL_ADC_PollForConversion(&hadc, 5);
-    // adc_7 = HAL_ADC_GetValue(&hadc);
-
-    // //last ADC in the sequence
-    // HAL_ADC_Start(&hadc);
-    // HAL_ADC_PollForConversion(&hadc, 5);
-    // adc_9 = HAL_ADC_GetValue(&hadc);
-  }
 
-  for(int i = 0; i < DSP_CONSTANTS::DMA_BUFFER_LENGTH; i++){
-    // for(int i = 0; i < DSP_CONSTANTS::WORKING_BUFFER_LENGTH; i++){
-    //   input[2*i] = ADC1ConvertedValues[i];
-    //   input[2*i+1] = 0; //Imaginary part set to zero
-    // }
-    /**
-     * @warning MUST BE UPDATED!!
-     */
-    data_hyd_port[i] = 0;
-    data_hyd_starboard[i] = 0;
-    data_hyd_stern[i] = 0;
+/**
+ * @brief Read the ADC by using DMA
+ * 
+ * @param p_data_hyd_port Pointer to the memory for the port hydrophone
+ * @param p_data_hyd_starboard Pointer to the starboard hydrophone's memory
+ * @param p_data_hyd_stern Pointer to the stern hydrophone's memory
+ * 
+ * @warning Unsure if I have read/understood it correctly!
+ * Assuming that there exists two possibilities for the DMA to push the
+ * data depending on the rank:
+ * 
+ *    1) 
+ *      Data is pushed in serial after rank. Since the DMA/ADC is in scan
+ *      mode, it scans all of the channel after rank. Most likely
+ *        
+ *        r1 = rank1, r2 = rank2, r3 = rank3
+ *        {r1, r2, r3, r1, r2, r3, ... , r1, r2, r3}
+ * 
+ *    2) 
+ *      All of the data for the highest priority rank is pushed first, then
+ *      the data belonging to the second priority rank and so forth
+ * 
+ *      {r1, r1, ..., r1, r2, r2, ..., r2, r3, r3, ..., r3}
+ * 
+ * From my understanding of the circular DMA and ADC in scan-mode, it uses
+ * the first method. This is assumed true, however that is potentially a serious
+ * bug! 
+ */
+static void read_ADC(float32_t* p_data_hyd_port, float32_t* p_data_hyd_starboard,
+            float32_t* p_data_hyd_stern){
+  /**
+   * Reading the data. Dropping the last couple of datapoints, since
+   * 4096 % 3 = 1. Reducing the number of datapoints reduces the 
+   * accuracy of the analysis, however prevents out-of-range error
+   */
+  for(int i = 0; i < DSP_CONSTANTS::DMA_BUFFER_LENGHT - 
+        NUM_HYDROPHONES, i += NUM_HYDROPHONES){
```

Comment:
Should be a ; instead of a , after NUM_HYDROPHONES

Reply:
Corrected. Thanks for noticing such a small typo